src/Eloquent/Model.php

```diff
@@ -284,23 +278,25 @@ public function setAttribute($key, $value)
     /** @inheritdoc */
     protected function asDecimal($value, $decimals)
     {
-        try {
-            $value = (string) BigDecimal::of((string) $value)->toScale((int) $decimals, RoundingMode::HALF_UP);
-
-            return new Decimal128($value);
-        } catch (BrickMathException $e) {
-            throw new MathException('Unable to cast value to a decimal.', previous: $e);
+        if ($value instanceof Decimal128) {
```

Comment:
This may convert `MongoDB\BSON\Int64` also.

Reply:
I will implement and add a test for this later :)